server/src/main/java/org/elasticsearch/common/bytes/BytesReference.java

```diff
@@ -53,6 +53,19 @@ static BytesReference bytes(XContentBuilder xContentBuilder) {
         }
     }
 
+    /**
+     * Convert an {@link XContentBuilder} into a BytesReference.
+     * This method doesn't close the builder,so it can be re-used.
+     */
```

Comment:
copy of `BytesReference.bytes` without `xContentBuilder.close()`. Tbh, I thought that closing builder is fine as even it closes stream it's safe - closing `BytesStreamOutput` is no-op anyway.

However, run into NPE when uses regular `bytes` call in this branch. Apparently it also closes internal generator.

Reply:
yep, actually don't even need a new method just do `stream.bytes`